subxt/src/lib.rs

```diff
@@ -168,46 +168,29 @@ pub mod ext {
 ///
 /// ## `substitute_type(path = "...", with = "...")`
 ///
-/// This attribute replaces any reference to the generated type at the path given by `path` with a
-/// reference to the path given by `with`.
+/// Replaces references to a specific type in the generated code with a custom type.
+///
+/// ### Important Notes
+/// - The path to replace must be an absolute path (fully qualified)
+/// - The replacement type must implement the necessary traits for encoding and decoding
+///
+/// ### Basic Usage
+/// Replace a specific type with a custom type:
 ///
 /// ```rust
 /// #[subxt::subxt(
 ///     runtime_metadata_path = "../artifacts/polkadot_metadata_full.scale",
 ///     substitute_type(path = "sp_arithmetic::per_things::Perbill", with = "crate::Foo")
 /// )]
-/// mod polkadot {}
-///
-/// # #[derive(
-/// #     scale_encode::EncodeAsType,
-/// #     scale_decode::DecodeAsType,
-/// #     codec::Encode,
-/// #     codec::Decode,
-/// #     Clone,
-/// #     Debug,
-/// # )]
-/// // In reality this needs some traits implementing on
-/// // it to allow it to be used in place of Perbill:
-/// pub struct Foo(u32);
-/// # impl codec::CompactAs for Foo {
-/// #     type As = u32;
-/// #     fn encode_as(&self) -> &Self::As {
-/// #         &self.0
-/// #     }
-/// #     fn decode_from(x: Self::As) -> Result<Self, codec::Error> {
-/// #         Ok(Foo(x))
-/// #     }
-/// # }
-/// # impl From<codec::Compact<Foo>> for Foo {
-/// #     fn from(v: codec::Compact<Foo>) -> Foo {
-/// #         v.0
-/// #     }
-/// # }
-/// # fn main() {}
+/// mod polkadot {
+///     // Custom type to replace Perbill
+///     pub struct Foo(u32);
+///     // Note: Foo must implement required traits
+/// }
```

Comment:
This looks wrong to me offhand; the substititue path is "crate::Foo" but then we move the type into `crate::polkadot::Foo`? I'd just keep the type defined at the top level to avoid anybody thinking that there is a requirement for the extra mod or whatever.